providers/alibaba/src/airflow/providers/alibaba/cloud/hooks/maxcompute.py

```diff
@@ -0,0 +1,250 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import functools
+from typing import TYPE_CHECKING, Any, Callable, TypeVar
+
+from odps import ODPS
+
+from airflow.exceptions import AirflowException
+from airflow.providers.alibaba.cloud.hooks.base_alibaba import AlibabaBaseHook
+
+if TYPE_CHECKING:
+    from odps.models import Instance
+
+RT = TypeVar("RT")
+
+
+def fallback_to_default_project_endpoint(func: Callable[..., RT]) -> Callable[..., RT]:
+    """
+    Provide fallback for MaxCompute project and endpoint to be used as a decorator.
+
+    If the project or endpoint is None it will be replaced with the project from the
+    connection extra definition.
+
+    :param func: function to wrap
+    :return: result of the function call
+    """
+
+    @functools.wraps(func)
+    def inner_wrapper(self, *args, **kwargs) -> RT:
+        if args:
+            raise AirflowException("You must use keyword arguments in this methods rather than positional")
+        if "project" in kwargs:
+            kwargs["project"] = kwargs["project"] or self.project
+        else:
+            kwargs["project"] = self.project
```

Comment:
```suggestion
        kwargs["project"] = kwargs.get("project", self.project)
```

does this work better?